### PR TITLE
feat: add `orientation` to divider

### PR DESCRIPTION
**Pull request by eljefe223**
# Pull Request

## 📖 Description
Adds `orientation` to `divider`

image

### 🎫 Issues
Closes: #5492 

### General

- [x] I have included a change request file using `$ yarn change`
- [x] I have added tests for my changes.
- [x] I have tested my changes.
- [x] I have updated the project documentation to reflect my changes.
- [x] I have read the [CONTRIBUTING](https://github.com/Microsoft/fast/blob/master/CONTRIBUTING.md) documentation and followed the [standards](https://www.fast.design/docs/community/code-of-conduct/#our-standards) for this project.

### Component-specific

- [ ] I have added a new component
- [x] I have modified an existing component
- [ ] I have updated the [definition file](https://github.com/Microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#definition)
- [x] I have updated the [configuration file](https://github.com/Microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#configuration)

### CODE REVIEW

File: packages/web-components/fast-components/src/divider/divider.styles.ts [hunk 1]
```diff
@@ -25,4 +25,12 @@ export const dividerStyles: (
             border: none;
             border-top: calc(${strokeWidth} * 1px) solid ${neutralStrokeDividerRest};
         }
+
+        :host([orientation="vertical"]) {
+            height: 100%;
+            margin: 0 calc(${designUnit} * 1px);
+            border: none;
```
Comment (radium-v) [inline packages/web-components/fast-components/src/divider/divider.styles.ts:32]: `border: none` is inherited from the cascade and isn't needed here.

Comment (eljefe223) [reply packages/web-components/fast-components/src/divider/divider.styles.ts:32]: I guess it can we removed from the existing style also then

**Review by radium-v — APPROVED** [review]

File: packages/web-components/fast-components/src/divider/divider.styles.ts [hunk 2]
```diff
@@ -22,7 +22,14 @@ export const dividerStyles: (
...
             height: 0;
             margin: calc(${designUnit} * 1px) 0;
-            border: none;
             border-top: calc(${strokeWidth} * 1px) solid ${neutralStrokeDividerRest};
+            border-left: none;
+        }
+
+        :host([orientation="vertical"]) {
+            height: 100%;
+            margin: 0 calc(${designUnit} * 1px);
```
Comment (chrisdholt) [inline packages/web-components/fast-components/src/divider/divider.styles.ts:31]: Is margin still necessary on either of these?

**Review by chrisdholt — APPROVED** [review]

File: packages/web-components/fast-components/src/divider/divider.styles.ts (hunk 2 above)
Comment (eljefe223) [reply packages/web-components/fast-components/src/divider/divider.styles.ts:31]: Yes, We discussed this in related conversations to #5407

**Pull request merged**